storage/src/mmr/mutable.rs

```diff
@@ -237,212 +324,344 @@ impl<K: Array, V: Array, H: CHasher> MutableMmr<K, V, H> {
 
         let digests = ops
             .iter()
-            .map(|op| MutableMmr::op_digest(hasher, op))
+            .map(|op| MutableMmr::<_, E, _, _, _>::op_digest(hasher, op))
             .collect::<Vec<_>>();
 
         proof.verify_range_inclusion(hasher, &digests, start_pos, end_pos, root_hash)
     }
 
-    /// Consume the store and return a [StoreState] from which the current active state of the store
-    /// can be fully recovered and fully proven.
-    pub fn to_state(mut self) -> StoreState<K, V, H> {
-        self.prune_known_inactive();
+    /// Commit any pending operations to the store, ensuring they are persisted to disk &
+    /// recoverable upon return from this function.
+    pub async fn commit(&mut self, hasher: &mut H) -> Result<(), Error> {
+        // Raise the inactivity floor by the # of uncommitted operations, plus 1 to account for the
+        // floor op that will be appended.
+        self.raise_inactivity_floor(hasher, self.uncommitted_ops + 1)
+            .await?;
+        self.uncommitted_ops = 0;
+        self.sync().await?;
+
+        // TODO: Make the frequency with which we prune known inactive items configurable in case
+        // this turns out to be a significant part of commit overhead.
+        self.prune_known_inactive().await
+    }
 
-        let prune_to_pos = leaf_num_to_pos(self.pruned_loc);
-        let pinned_nodes = self.ops.node_digests_to_pin(prune_to_pos);
+    /// Sync the store to disk ensuring the current state is persisted.
+    async fn sync(&mut self) -> Result<(), Error> {
+        futures::try_join!(
```

Comment:
nit: use `try_join!` directly (given you already use `try_join_all` directly)

Reply:
done